libopenage/pathfinding/integrator.cpp

```diff
@@ -97,25 +105,32 @@ std::shared_ptr<FlowField> Integrator::build(const std::shared_ptr<IntegrationFi
                                              const std::shared_ptr<IntegrationField> &other,
                                              sector_id_t other_sector_id,
                                              const std::shared_ptr<Portal> &portal,
-                                             bool with_los) {
+                                             bool with_los,
+                                             bool evict_cache) {
 	auto cache_key = std::make_pair(portal->get_id(), other_sector_id);
-	auto cached = this->field_cache.find(cache_key);
-	if (cached != this->field_cache.end()) {
+	if (evict_cache) {
+		this->field_cache->evict(cache_key);
+	}
```

Comment:
same here

Reply:
The integrate function does not pass in the cost field. I am going to remove the dirty check from the integrate function for now, unless we want to add dirty flags to the integration fields as well.

Reply:
`integrate` receives `const std::shared_ptr<CostField> &cost_field` as its first argument, so you can use it for checking whether the field is dirty.

Reply:
sorry, I meant build()